src/components/CTreeview/CTreeview.js

```diff
@@ -0,0 +1,303 @@
+            }
+          });
+        };
+        searchTree(this.items);
+        return 'id';
+      }
+      return null;
+    },
+    itemDisplay() {
+      if (this.items.length) {
+        // User-defined key
+        const itemDisplay = this.config.itemDisplay;
+        if (!isNil(itemDisplay) && !isEmpty(itemDisplay)) {
+          return itemDisplay;
+          // Predefined 'name' key
+        } else if (!isNil(this.items[0].name)) {
+          return 'name';
+        }
+        // Find first string key
+        let firstChild = null;
+        const filtered = item => each(item, (value, key) => {
+          if (isString(value)) {
+            firstChild = key;
+          }
+          if (isObject(value) && isNil(firstChild)) filtered(value);
+        });
+        filtered(this.items);
+        return firstChild;
+      }
+      return null;
+    },
+    openOnLoad() {
+      let openState = false;
+      this.open = [];
+      switch (this.config.defaultState) {
+        case 'first':
+          if (this.items.length > 0) {
+            this.open.push(this.firstItem[this.itemValue]);
+          }
+          openState = false;
+          break;
+        case 'all':
+          // Since openAll is only working on component load
+          // set open value trought array (this.open) of item id's
+          if (this.items.length > 0) {
+            this.open = this.allItems;
+          }
+          openState = true;
+          break;
+        default:
+          openState = false;
+      }
+      return openState;
+    },
+    dataSourceSchema() {
+      if (this.dataSource) return this.dataSource.schema;
+      return null;
+    },
+    togglePosition() {
+      return this.config.expanderPosition === 'right' ? 'c-right-toggle' : '';
+    },
+    checkPosition() {
+      return this.config.selectorPosition === 'right' ? 'c-right-check' : '';
+    },
+  },
+  methods: {
+    addItem() {
+      // TODO
+    },
+    getMapType() {
+      const type = filter(this.dataSourceSchema, (schema) => {
+        const name = schema.mapName || schema.name;
+        return name === this.itemDisplay;
+      });
+      return type.length ? type[0].mapType : null;
+    },
+    // Get all children id's (open state is controlled by this)
+    getAllItems() {
+      const leafs = [];
+      const searchTree = (items) => {
+        each(items, (child) => {
+          if (child[this.itemChildren] && child[this.itemChildren].length > 0) {
+            searchTree(child[this.itemChildren]);
+          }
+          if (child[this.itemValue]) leafs.push(child[this.itemValue]);
+        });
+      };
+      searchTree(this.items);
+      return sortBy(leafs);
+    },
+    loadData() {
+      this.loadConnectorData().then((result) => {
+        this.items = result.items || [];
+        this.sendToEventBus('DataSourceChanged', this.dataSource);
+      });
+    },
+    async getChildren(item) {
+      this.sendToEventBus('GetAsyncChildren', item);
```

Comment:
Everything seems ok, except for this part.
We should not emit here, but load children using connector in dataSource property (if exists).